controlplane/kubeadm/api/v1beta1/kubeadmcontrolplanetemplate_webhook.go

```diff
@@ -27,12 +27,19 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 
 	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
+	clusteradmission "sigs.k8s.io/cluster-api/exp/admission"
 	"sigs.k8s.io/cluster-api/feature"
 )
 
 const kubeadmControlPlaneTemplateImmutableMsg = "KubeadmControlPlaneTemplate spec.template.spec field is immutable. Please create new resource instead."
 
 func (r *KubeadmControlPlaneTemplate) SetupWebhookWithManager(mgr ctrl.Manager) error {
```

Comment:
I think we only have to do this for templates that we rotate. For regular objects it's not relevant as the dryrun SSA call is identical to what we do later on. So any validation error aligns with the one we would otherwise get with SSA later.

Quick overview:
* no rotation
    * KubeadmControlPlaneTemplate (ControlPlaneTemplate) is reconciled to KubeadmControlPlane
    * DockerClusterTemplate (InfrastructureClusterTemplate) is reconciled to DockerCluster
* rotation
    * DockerMachineTemplate (InfrastructureMachineTemplate) is reconciled to DockerMachineTemplate
    * KubeadmConfigTemplate (BootstrapConfigTemplate) is reconciled to KubeadmConfigTemplate

=> As KubeadmConfigTemplate doesn't have immutability checks we only have to do it for DockerMachineTemplate